erpnext/stock/doctype/warehouse/test_warehouse.py

```diff
@@ -92,6 +93,40 @@ def test_warehouse_merging(self):
 		self.assertTrue(frappe.db.get_value("Warehouse",
 			filters={"account": "Test Warehouse for Merging 2 - TCP1"}))
 
+	def test_unlinking_warehouse_from_item_defaults(self):
+		company = "_Test Company"
+
+		warehouse_names = [f'_Test Warehouse {i} for Unlinking' for i in range(2)]
+		warehouse_ids = []
+		for warehouse in warehouse_names:
+			warehouse_id = create_warehouse(warehouse, company=company)
+			warehouse_ids.append(warehouse_id)
+
+		item_names = [f'_Test Item {i} for Unlinking' for i in range(2)]
+		for item in item_names:
+			create_item(item)
+			assign_default_warehouses_to_item(item, warehouse_ids, company)
+
+		# Delete warehouses
+		for warehouse in warehouse_ids:
+			frappe.delete_doc("Warehouse", warehouse)
+
+		# Check Item existance
+		for item in item_names:
+			item_exists = frappe.db.exists("Item", item)
+			self.assertTrue(item_exists, item)
```

Comment:
This will become somewhat flaky because of how frappe.db.exists is defined. Maybe cast it to `bool` first? Also second parameter is a message that will be shown on failure. Just `item` isn't that useful.